### ISSUE TITLE
Combined usage of `--deviceId` and `--port` is broken.

### ISSUE THREAD

**Issue by cortinico** [react-native-community/cli#1792]
## Environment

```
System:
    OS: macOS 13.0.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 619.30 MB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.14.2 - /var/folders/j4/3ws5dkhj3l73hwzrv3sh709c0000gn/T/yarn--1673606188118-0.5188258691218024/node
    Yarn: 1.22.18 - /var/folders/j4/3ws5dkhj3l73hwzrv3sh709c0000gn/T/yarn--1673606188118-0.5188258691218024/yarn
    npm: 9.2.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
    Watchman: 20230107 - /usr/local/bin/watchman
  Managers:
    CocoaPods: Not Found
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 16.0, macOS 12.3, tvOS 16.0, watchOS 9.0
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8512546
    Xcode: 14.0/14A309 - /usr/bin/xcodebuild
  Languages:
    Java: 18.0.1 - /Users/ncor/.jenv/shims/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0
    react-native: 0.71.0 => 0.71.0
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

## Description

I've just noticed that invoking the following:

```
yarn react-native run-android --deviceId emulator-5554 --port 8082
```

to run Android on a given emulator with a specific port is broken. Specifically, the port parameter is ignored, and fallback to `8081`.
I'm using this command as I have two emulators and I want to run 2 distinct project in parallel (with two instances of metro on different ports).

If I kill one of the two emulator and invoke just:
```
yarn react-native run-android --port 8082
```
everything seems to work fine.

## Reproducible Demo

1. Create a new project with `npx react-native init RN071`
2. Start an Android emulator and get it's name with `adb devices`.
3. Run `yarn react-native run-android --deviceId emulator-5554 --port 8082` => Verify that it does **not** connect to Metro
4. Run `yarn react-native run-android --port 8082` => Verify that it does connect to Metro

**Comment by szymonrybczak** [comment]
Hey @cortinico, this issue was recently fixed by https://github.com/react-native-community/cli/pull/1785 😄

**Comment by cortinico** [comment]
> Hey @cortinico, this issue was recently fixed by #1785 😄

Thanks for the heads up, so can we close this then?

**Comment by szymonrybczak** [comment]
Yup! 😄
